src/test/scala/org/broadinstitute/dsde/firecloud/service/NotificationsApiServiceSpec.scala

```diff
@@ -22,23 +18,29 @@ class NotificationsApiServiceSpec extends FlatSpec with BeforeAndAfterAll with H
     MockWorkspaceServer.stopWorkspaceServer()
   }
 
-  "NotificationsApiService" should "get workspace notifications" in {
-    Get(s"/api/notifications/workspace/${MockWorkspaceServer.mockValidWorkspace.namespace}/${MockWorkspaceServer.mockValidWorkspace.name}") ~> dummyAuthHeaders ~>
-      sealRoute(notificationsRoutes) ~>
-      check {
-        assertResult(StatusCodes.OK, response.entity.asString) {
-          status
-        }
+  "NotificationsApiService" - {
+    "get workspace notifications" in {
+      val namespace = MockWorkspaceServer.mockValidWorkspace.namespace
+      val name = MockWorkspaceServer.mockValidWorkspace.name
+      val workspaceNotificationUri = s"/api/notifications/workspace/$namespace/$name"
+
+      doAssert(GET, workspaceNotificationUri, OK)
+    }
+
+    "get general notifications" in {
+      doAssert(GET, "/api/notifications/general", OK)
+    }
+
+    "non-GET methods should be rejected" in {
+      allHttpMethodsExcept(GET) foreach { method =>
+        doAssert(method, "/api/notifications/general", MethodNotAllowed)
       }
+    }
   }
 
-  it should "get general notifications" in {
-    Get(s"/api/notifications/general") ~> dummyAuthHeaders ~>
-      sealRoute(notificationsRoutes) ~>
-      check {
-        assertResult(StatusCodes.OK, response.entity.asString) {
-          status
-        }
-      }
+  private def doAssert(method: HttpMethod, uri: String, response: StatusCode): Unit = {
+    new RequestBuilder(method)(uri) ~> dummyAuthHeaders ~> sealRoute(notificationsRoutes) ~> check {
+      status should be(response)
```

Comment:
nit: avoid using "response" as a variable name (or "request" or "context" and almost certainly others). Various spray libraries create those variable names implicitly. How about `expectedStatus`?